### PR TITLE
lambda-promtail: Add support for getting loki credentials with secret…

### PR DESCRIPTION
**Pull request by lfelipevaca**
… manager

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Documentation added
- [ ] Tests updated
- [x] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Comment by lfelipevaca** [comment]
this change implements the feature requested in the following issue

https://github.com/grafana/loki/issues/7169

File: tools/lambda-promtail/lambda-promtail/aws_secrets.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"encoding/json"
+	"fmt"
+)
+
+
+func getSecret(secretName,usernameKey, passwordKey, region string) (username, password string) {
```
Comment (MichelHollands) [inline tools/lambda-promtail/lambda-promtail/aws_secrets.go:16]: Can you rename this to `getAWSSecret()`? It's clearer that it's only for AWS that way.

File: tools/lambda-promtail/lambda-promtail/aws_secrets.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	//Create a Secrets Manager client
```
Comment (MichelHollands) [inline tools/lambda-promtail/lambda-promtail/aws_secrets.go:17]:
```suggestion
	// Create a Secrets Manager client
```

File: tools/lambda-promtail/lambda-promtail/main.go [hunk 3]
```diff
@@ -55,8 +55,13 @@ func setupArguments() {
 		panic(err)
 	}
 
-	username = os.Getenv("USERNAME")
-	password = os.Getenv("PASSWORD")
+	if (os.Getenv("SECRET_NAME") != "" && os.Getenv("SECRET_USERNAME_KEY") != "" && os.Getenv("SECRET_PASSWORD_KEY") != "" && os.Getenv("AWS_REGION") != "") {
```
Comment (MichelHollands) [inline tools/lambda-promtail/lambda-promtail/main.go:58]: Can you add a comment saying this is for AWS?

File: tools/lambda-promtail/lambda-promtail/main.go [hunk 4]
```diff
@@ -55,8 +55,13 @@ func setupArguments() {
...
+		username, password = getSecret(os.Getenv("SECRET_NAME"),os.Getenv("SECRET_USERNAME_KEY"),os.Getenv("SECRET_PASSWORD_KEY"),os.Getenv("AWS_REGION"))
+	}else{
```
Comment (MichelHollands) [inline tools/lambda-promtail/lambda-promtail/main.go:60]:
```suggestion
	} else {
```

**Comment by MichelHollands** [comment]
@lfelipevaca Thanks for your contribution. Can you check the review comments and rebase? It's looking fine otherwise.

**Comment by jeschkies** [comment]
@lfelipevaca thanks for your contribution. Would you mind addressing Michel's comments?

**Comment by chaudum** [comment]
@lfelipevaca Are you still interested in getting this done? The lambda-promtail code change quite a bit in the meantime. Would you mind rebasing your branch and addressing @MichelHollands 's comments?

**Comment by MasslessParticle** [comment]
I'm closing this one due to inactivity.

**Comment by SSE69** [comment]
Hello, I'm interested in this change. Would it be possible to reopen it so that I can finish what's required in the PR?

**Pull request closed without merging**